src/weather_forecast.py

```diff
@@ -0,0 +1,222 @@
+import streamlit as st
+import pandas as pd
+import plotly.express as px
+from herbie import FastHerbie
+from datetime import datetime, timedelta, time
+import numpy as np
+from concurrent.futures import ThreadPoolExecutor
+from typing import List, Optional, Tuple
+
+@st.cache_data(show_spinner=False)
+def fetch_data_for_init_time(
+    init_time: datetime, 
+    forecast_date: datetime, 
+    lat: float, 
+    lon: float, 
+    parameter: str, 
+    model: str = "ifs"
+) -> Tuple[Optional['xarray.Dataset'], Optional[datetime]]:
+    """
+    Fetch the weather forecast data for a specific initialization time and forecast date.
+
+    Parameters:
+    - init_time (datetime): The initialization time (usually yesterday at a specific hour).
+    - forecast_date (datetime): The forecast date for which to get the forecast.
+    - lat (float): Latitude of the location for which the forecast is requested.
+    - lon (float): Longitude of the location for which the forecast is requested.
+    - parameter (str): The weather parameter to retrieve. Examples: "u10:v10", "u100:v100", or "2t".
+    - model (str): The weather model to use for fetching data (default: "ifs").
+
+    Returns:
+    - ds (xarray.Dataset or None): The forecast dataset for the specified parameters.
+    - init_time (datetime or None): The initialization time for the dataset.
+    """
+    # Adjust the initialization datetime
+    FH = FastHerbie([init_time], model=model, fxx=range(init_time.hour, 24 + init_time.hour, 1), fast=True)
```

Comment:
housldnt this be `(forecast_date - init_time).hours` or something like that

Reply:
i have already set up the init times to be forecast day-1. So to give init times of yesterday for the forecast of today.